pkg/util/pki/keyusage.go

```diff
@@ -148,3 +150,23 @@ func buildASN1KeyUsageRequest(usage x509.KeyUsage) (pkix.Extension, error) {
 
 	return OIDExtensionKeyUsage, nil
 }
+
+// Allow users of the API to pass in an OID string and we will convert to an asn1.ObjectIdentifier
+// https://datatracker.ietf.org/doc/html/rfc4519.html
+func ToAsn1Oid(oidString string) (asn1.ObjectIdentifier, error) {
```

Comment:
We should perform all this kind of validation with a regex pattern on the CRD to avoid having any unparseable/invalid data 😊

Reply:
Hm - that's an interesting idea. I didn't even know you could do that. I just looked it up and it looks like you can use the "pattern" field to do this? I'll give it a shot, thanks.